src/tools/tidy/src/error_codes_check.rs

```diff
@@ -50,41 +52,75 @@ const WHITELIST: &[&str] = &[
     "E0729",
 ];
 
-fn extract_error_codes(f: &str, error_codes: &mut HashMap<String, bool>) {
+fn check_error_code_explanation(
+    f: &str,
+    error_codes: &mut HashMap<String, bool>,
+    err_code: String,
+) {
+    for line in f.lines() {
+        let s = line.trim();
+        if s.starts_with("```") && s.contains("compile_fail") && s.contains('E') {
+            let nb = error_codes.entry(err_code).or_insert(false);
+            *nb = true;
```

Comment:
What is the point of this `or_insert(false)` and then always setting it to true? Why not just `insert(err_code, true)`?

Reply:
Good point! Updating.